openapi/cue/genoapi.go

```diff
@@ -0,0 +1,446 @@
+	"cuelang.org/go/cue/format"
+	"cuelang.org/go/cue/load"
+	"cuelang.org/go/encoding/openapi"
+	"cuelang.org/go/encoding/protobuf"
+	"github.com/emicklei/proto"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+var (
+	configFile = flag.String("f", "", "configuration file; by default the directory  in which this file is located is assumed to be the root")
+
+	inplace = flag.Bool("inplace", false, "generate configurations in place")
+	paths   = flag.String("paths", "/protobuf", "comma-separated path to search for .proto imports")
+	verbose = flag.Bool("v", false, "print debugging output")
+
+	// manually configuring builds
+	all = flag.Bool("all", false, "combine all the matched outputs in a single file; the 'all' section must be specified in the configuration")
+)
+
+func main() {
+	flag.Parse()
+	log.SetFlags(log.Lshortfile)
+
+	cwd, _ := os.Getwd()
+
+	// Include paths for proto evaluation. Take relative paths relative to the
+	// current working directory.
+	importPaths := []string{cwd} // change cwd later if root is set.
+	for _, p := range strings.Split(*paths, ",") {
+		if p != "" {
+			abs, _ := filepath.Abs(p)
+			importPaths = append(importPaths, abs)
+		}
+	}
+
+	if *configFile == "" {
+		log.Fatalf("Must specify configuration with the -f option")
+	}
+
+	c, err := loadConfig(*configFile)
+	if err != nil {
+		fatal(err, "Error parsing configuration file")
+	}
+
+	dir, err := filepath.Abs(*configFile)
+	if err != nil {
+		log.Fatalf("Invalid root %q: %v", dir, err)
+	}
+	cwd, _ = filepath.Split(dir)
+
+	importPaths[0] = cwd
+
+	b := protobuf.NewExtractor(&protobuf.Config{
+		Root:   cwd,
+		Module: c.Module,
+		Paths:  importPaths,
+	})
+
+	_ = filepath.Walk(cwd, func(path string, f os.FileInfo, _ error) (err error) {
+		if !strings.HasSuffix(path, ".proto") {
+			return nil
+		}
+		return b.AddFile(path, nil)
+	})
+
+	files, err := b.Files()
+	if err != nil {
+		fatal(err, "Error generating CUE from proto")
+	}
+
+	modRoot := cwd
+	if !*inplace {
+		modRoot, err = ioutil.TempDir("", "genoapi")
+		if err != nil {
+			log.Fatalf("Error creating temp dir: %v", err)
+		}
+		defer os.RemoveAll(modRoot)
+	}
+
+	c.cwd = cwd
+	c.modRoot = modRoot
+
+	for _, f := range files {
+		b, err := format.Node(f)
+		if err != nil {
+			fatal(err, "Error formatting file: ")
+		}
+		filename := f.Filename
+		relPath, _ := filepath.Rel(cwd, filename)
+		filename = filepath.Join(modRoot, relPath)
+
+		_ = os.MkdirAll(filepath.Dir(filename), 0755)
+		if err := ioutil.WriteFile(filename, b, 0644); err != nil {
+			log.Fatalf("Error writing file: %v", err)
+		}
+	}
+
+	// Gernate the OpenAPI
+	protoNames := map[string]string{}
+
+	// build map of CUE package import paths (Go paths) to proto package paths.
+	builds, _ := b.Instances()
+	for _, i := range builds {
+		protoNames[i.ImportPath] = i.DisplayPath
+	}
+
+	builder := &builder{
+		Config:     c,
+		protoNames: protoNames,
+	}
+
+	// Build the OpenAPI files.
+	if *all {
+		if c.All == nil {
```

Comment:
if the "All" section is specified, it is a redundant to also specify it in the flag?

Reply:
All is only generated when the --all flag is specified. You mentioned the aggregate file is only needed in some cases. But perhaps you want to solve this by having two different configuration files?